### PR TITLE
Update to python 3

### PR DESCRIPTION
**Pull request by JamesJeffryes**

### CODE REVIEW

**Comment by kkellerlbl** [comment]
@JamesJeffryes can you make a PR against the develop branch instead?

Is it expected that Travis tests fail?

**Comment by JamesJeffryes** [comment]
@kkellerlbl Yeah, I think it fails because it is a PR from my fork and thus travis can't decrypt the token for NMS correctly. I think it might work when merging from dev to master on the same repo though

File: lib/biokbase/catalog/Client.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 try:
     # baseclient and this client are in a package
     from .baseclient import BaseClient as _BaseClient  # @UnusedImport
-except:
+except ImportError:
```
Comment (kkellerlbl) [inline lib/biokbase/catalog/Client.py]: Is it safe to assume the changes in Client.py are just new autogenerated code from the SDK?

File: lib/biokbase/catalog/Impl.py [hunk 2]
```diff
@@ -20,9 +20,9 @@ class Catalog:
     # state. A method could easily clobber the state set by another while
     # the latter method is running.
     ######################################### noqa
-    VERSION = "2.1.2"
-    GIT_URL = "https://github.com/JamesJeffryes/catalog.git"
-    GIT_COMMIT_HASH = "82480c4f938d5d1d5a85668a0ebaf1f27bf793f3"
+    VERSION = "0.0.1"
```
Comment (kkellerlbl) [inline lib/biokbase/catalog/Impl.py]: The version is probably not 0.0.1

File: lib/biokbase/catalog/Server.py [hunk 3]
```diff
@@ -1,23 +1,29 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-from wsgiref.simple_server import make_server
-import sys
+import datetime
```
Comment (kkellerlbl) [inline lib/biokbase/catalog/Server.py]: Also autogenerated SDK code?

File: lib/biokbase/catalog/controller.py [hunk 4]
```diff
@@ -721,7 +671,7 @@ def list_local_functions(self, params):
         else:
             release_tag = 'release'
         if 'release_tag' in params:
-            if not isinstance(params['release_tag'],basestring):
+            if not isinstance(params['release_tag'],str):
```
Comment (kkellerlbl) [inline lib/biokbase/catalog/controller.py]: Are these changes from 'basestring' to 'str' important?

File: lib/biokbase/catalog/version.py [hunk 5]
```diff
@@ -1,2 +1,2 @@
 # File that simply defines version information
-CATALOG_VERSION = '2.1.3'
\ No newline at end of file
+CATALOG_VERSION = '2.2.0'
```
Comment (kkellerlbl) [inline lib/biokbase/catalog/version.py]: The version here seems ok.  Maybe the version in Impl.py is irrelevant?

Comment (JamesJeffryes) [reply lib/biokbase/catalog/version.py]: Yup

File: lib/biokbase/catalog/Impl.py (hunk 2 above)
Comment (JamesJeffryes) [reply lib/biokbase/catalog/Impl.py]: it's not registered in the catalog (so meta) so the version in impl is irrelevant

File: lib/biokbase/catalog/Client.py (hunk 1 above)
Comment (JamesJeffryes) [reply lib/biokbase/catalog/Client.py]: Yeah, also the changes to authclient and baseclient

File: lib/biokbase/catalog/Server.py (hunk 3 above)
Comment (JamesJeffryes) [reply lib/biokbase/catalog/Server.py]: Yup

File: lib/biokbase/catalog/controller.py (hunk 4 above)
Comment (JamesJeffryes) [reply lib/biokbase/catalog/controller.py]: basestring is no more in Python 3

**Comment by JamesJeffryes** [comment]
@kkellerlbl I'd be happy to give that a try (you could also give me write access to the repo and just protect the master branch from direct commits and unreviewed PRs in the settings).

**Comment by kkellerlbl** [comment]
@JamesJeffryes let's try your second suggestion.  I've given you write access, and (I hope) configured master as protected properly.

**Pull request merged**